src/scanner.rs

```diff
@@ -42,6 +57,12 @@ impl ScanConfig {
         self
     }
 
+    #[inline]
+    pub fn extend_filters(mut self, filters: Vec<Filter>) -> Self {
```

Comment:
I like this idea of using an enum for different filters. This interface should preferably accept a slice, for example:
```rust
with_filters(mut self, filters: &[Filter]) -> Self
```